toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -615,47 +622,92 @@ func (r *RpmRepoCloner) clonePackage(baseArgs []string) (preBuilt bool, err erro
 
 		finalArgs := append(baseArgs, reposArgs...)
 
-		var (
-			stdout string
-			stderr string
-		)
-		stdout, stderr, err = shell.Execute("tdnf", finalArgs...)
+		// We run in a retry loop on errors deemed retriable.
+		cancel := make(chan struct{})
+		retryNum := 1
```

Comment:
I'm fine leaving `retryNum` here but if you have time, I think it might be worth considering changing how `Run*` functions work, so that they pass they keep track of the attempt number (print it on the debug level for instance) and the callback can then assume the attempt counting is done for them in the background.

Reply:
Nevermind, I gave it a shot as well and it looks this change would be too intrusive for this PR.